app/controllers/users_controller.rb

```diff
@@ -1,9 +1,19 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
+
+  def index
+    page_size = params[:per_page].present? ? params[:per_page] :
```

Comment:
dau em, minh setting the nay sau refactor code neu can mo rong thi se them 1 line code vao tat cac cac ham update param truyen vao thoi. Em lam API chua, client tùy vào máy mobie chậm yếu hoặc màn hình ngắn dài mà load pagesize thích hợp em ah